### PR TITLE
Fix GetAssessmentResult authz check

### PR DESCRIPTION
**Pull request by anatheka**
Only the "allowed" cloud_services were checked. The check for "all" was missing.

### CODE REVIEW

**Review by lebogg — APPROVED** [review]
Good catch. Maybe add a small test for the new case?

**Comment by oxisto** [comment]
> Good catch. Maybe add a small test for the new case?

Would be a good idea. I just extended the auth strategy mock to include the "all" flag in https://github.com/clouditor/clouditor/pull/1043. This makes it easier to test it.

**Comment by lebogg** [comment]
Not blocked anymore :+1:

File: service/orchestrator/assessment_results_test.go [hunk 1]
```diff
@@ -83,20 +82,23 @@ func TestService_GetAssessmentResult(t *testing.T) {
...
 				authz: servicetest.NewAuthorizationStrategy(false, []string{orchestratortest.MockAssessmentResult1.CloudServiceId}),
 			},
 			args: args{
-				req: orchestratortest.MockAssessmentResultRequest1,
+				req: orchestratortest.MockAssessmentResultRequest2,
+				ctx: testutil.TestContextOnlyService1,
+			},
+			res: nil,
+			wantErr: func(t assert.TestingT, err error, i ...interface{}) bool {
+				return assert.Contains(t, err.Error(), "access denied")
```
Comment (lebogg) [inline service/orchestrator/assessment_results_test.go:100]:
```suggestion
				return assert.ErrorContains(t, err, service.ErrPermissionDenied.Error())
```

Comment (lebogg) [reply service/orchestrator/assessment_results_test.go:100]: Would be more safe if, e.g., the message of `ErrPermissionDenied` would be changed in the future

**Review by lebogg — APPROVED** [review]
LGTM.

**Pull request merged**